backend/core/src/main/java/org/sonarsource/sonarlint/core/tracking/TrackingService.java

```diff
@@ -0,0 +1,304 @@
+      serverHotspot.getStatus().isResolved(), trackedHotspot.getSeverity(), RuleType.SECURITY_HOTSPOT, trackedHotspot.getRuleKey(),
+      trackedHotspot.getTextRangeWithHash(), trackedHotspot.getLineWithHash(),
+      serverHotspot.getKey(), trackedHotspot.getImpacts(), trackedHotspot.getFlows(), trackedHotspot.getQuickFixes(),
+      serverHotspot.getVulnerabilityProbability(), HotspotStatus.valueOf(serverHotspot.getStatus().name()), trackedHotspot.getRuleDescriptionContextKey(),
+      trackedHotspot.getCleanCodeAttribute(), trackedHotspot.getFileUri());
+  }
+
+  private static TrackedIssue updateTrackedIssueWithLocalOnlyIssueData(TrackedIssue trackedIssue, LocalOnlyIssue localOnlyIssue) {
+    return new TrackedIssue(trackedIssue.getId(), trackedIssue.getMessage(), trackedIssue.getIntroductionDate(),
+      localOnlyIssue.getResolution() != null, trackedIssue.getSeverity(), trackedIssue.getType(), trackedIssue.getRuleKey(), trackedIssue.getTextRangeWithHash(),
+      trackedIssue.getLineWithHash(), trackedIssue.getServerKey(), trackedIssue.getImpacts(), trackedIssue.getFlows(),
+      trackedIssue.getQuickFixes(), trackedIssue.getVulnerabilityProbability(), trackedIssue.getHotspotStatus(), trackedIssue.getRuleDescriptionContextKey(),
+      trackedIssue.getCleanCodeAttribute(), trackedIssue.getFileUri());
+  }
+
+  private MatchingSession startMatchingSession(String configurationScopeId, Iterable<ClientInputFile> files) {
+    var knownFindingsStore = knownFindingsStorageService.get();
+    var fileRelativePaths = StreamSupport.stream(files.spliterator(), false).map(ClientInputFile::relativePath).map(Path::of).collect(Collectors.toSet());
+    var issuesByRelativePath = fileRelativePaths.stream()
+      .collect(toMap(Function.identity(), relativePath -> knownFindingsStore.loadIssuesForFile(configurationScopeId, relativePath)));
+    var hotspotsByRelativePath = fileRelativePaths.stream()
+      .collect(toMap(Function.identity(), relativePath -> knownFindingsStore.loadSecurityHotspotsForFile(configurationScopeId, relativePath)));
+    var introductionDateProvider = getIntroductionDateProvider(configurationScopeId, fileRelativePaths);
+    var previousFindings = new KnownFindings(issuesByRelativePath, hotspotsByRelativePath);
+    return new MatchingSession(previousFindings, introductionDateProvider);
+  }
+
+  private IntroductionDateProvider getIntroductionDateProvider(String configurationScopeId, Set<Path> fileRelativePaths) {
+    var baseDir = getBaseDir(configurationScopeId);
+    if (baseDir != null) {
+      try {
+        var sonarLintBlameResult = GitBlameUtils.blameWithFilesGitCommand(baseDir, fileRelativePaths);
+        return (filePath, lineNumbers) -> determineIntroductionDate(filePath, lineNumbers, sonarLintBlameResult);
+      } catch (Exception e) {
+        LOG.error("Cannot access blame info for " + configurationScopeId, e);
+      }
+    }
+    return (filePath, lineNumber) -> null;
```

Comment:
We can return Instant.now() here right away.

Reply:
Good suggestion, it simplified indeed